src/utils/loose-equal.js

```diff
@@ -31,30 +31,32 @@ function looseEqual(a, b) {
   if (typeof a !== typeof b) {
     return false
   }
-  const validDatesCount = [isDate(a), isDate(b)].filter(Boolean).length
-  if (validDatesCount > 0) {
-    return validDatesCount === 2 ? a.getTime() === b.getTime() : false
+  let validTypesCount = [isDate(a), isDate(b)].filter(Boolean).length
+  if (validTypesCount > 0) {
+    return validTypesCount === 2 ? a.getTime() === b.getTime() : false
   }
-  const validFilesCount = [isFile(a), isFile(b)].filter(Boolean).length
-  if (validFilesCount > 0) {
-    return validFilesCount === 2 ? a === b : false
+  validTypesCount = [isFile(a), isFile(b)].filter(Boolean).length
+  if (validTypesCount > 0) {
+    return validTypesCount === 2 ? a === b : false
   }
-  const validArraysCount = [isArray(a), isArray(b)].filter(Boolean).length
-  if (validArraysCount > 0) {
-    return validArraysCount === 2
+  validTypesCount = [isArray(a), isArray(b)].filter(Boolean).length
+  if (validTypesCount > 0) {
+    return validTypesCount === 2
       ? a.length === b.length && a.every((e, i) => looseEqual(e, b[i]))
       : false
   }
```

Comment:
Could this just be:
```js
  if (validTypesCount === 2) {
    return a.length === b.length && a.every((e, i) => looseEqual(e, b[i]))
  }
```
and change all the `if`'s to `else if`? and let the final return false kick in?

Reply:
Err... never mind....

Would need to move the computation of the validTypesCount into the if condition